pkg/tarmak/provider/amazon/image.go

```diff
@@ -50,20 +51,54 @@ func (a *Amazon) QueryImages(tags map[string]string) (images []tarmakv1alpha1.Im
 			}
 		}
 
-		// TODO: determine whether image encrypted and set flag accordingly
-
 		creationTimestamp, err := time.Parse(formatRFC3339amazon, *ami.CreationDate)
 		if err != nil {
 			return images, fmt.Errorf("error parsing time stamp '%s'", err)
 		}
 		image.CreationTimestamp.Time = creationTimestamp
 		image.Name = *ami.ImageId
 		image.Location = a.Region()
-		images = append(
-			images,
-			image,
-		)
+
+		if ami.RootDeviceName == nil {
+			a.log.Warnf("failed to obtain root device name of ami '%s'", image.Name)
+			continue
+		}
+		rootName := *ami.RootDeviceName
+
+		foundRoot := false
+		for _, d := range ami.BlockDeviceMappings {
+			if d.DeviceName != nil && *d.DeviceName == rootName {
+				if d.Ebs == nil || d.Ebs.Encrypted == nil {
+					a.log.Warnf("failed to determine the encryption state of ami '%s'", image.Name)
+					continue
+				}
+
+				image.Encrypted = *d.Ebs.Encrypted
+				foundRoot = true
+				break
+			}
+		}
+
+		if !foundRoot {
+			a.log.Warnf("failed to find root device of ami '%s'", image.Name)
+			continue
+		}
+
+		images = append(images, image)
 	}
 
 	return images, nil
 }
+
+func (a *Amazon) verifyImageExists() error {
+	images, err := a.tarmak.Packer().List()
+	if err != nil {
+		return err
+	}
+
+	if len(images) == 0 {
+		return errors.New("no images found, please run `$ tarmak cluster images build`")
+	}
+
```

Comment:
we need to check the output of packer.IDs contains all instance pool's images. Ideally that should happen in the instance pool verify anyhow

Reply:
https://github.com/jetstack/tarmak/blob/master/pkg/tarmak/cluster/cluster.go#L616

https://github.com/jetstack/tarmak/blob/master/pkg/tarmak/cluster/cluster.go#L376